crates/local-cluster-runner/examples/three_nodes_and_metadata.rs

```diff
@@ -43,23 +43,24 @@ async fn main() {
     let shutdown_fut = shutdown();
 
     let mut cluster = cluster.start().await.unwrap();
+    let mut admin_output = cluster.nodes[0].lines(Regex::new("Server listening").unwrap());
 
-    match cluster.nodes[0]
-        .lines(Regex::new("Server listening").unwrap())
-        .next()
-        .await
-    {
-        None => {
-            error!("metadata node exited early");
-            std::process::exit(1)
-        }
-        Some(line) => {
-            info!("matched metadata logline: {line}")
-        }
-    };
+    tokio::select! {
+        _ = shutdown_fut => {}
 
-    shutdown_fut.await;
+        line = admin_output.next() => match line
+        {
+            None => {
+                error!("metadata node exited early");
+                std::process::exit(1)
+            }
+            Some(line) => {
+                info!("matched metadata logline: {line}")
+            }
+        }
+    }
```

Comment:
W/o waiting for the `shutdown_fut` after this `tokio::select` statement, we'll immediately terminate the cluster if the `admin_output.next()` arm has fired but not ctrl+c was pressed.

Reply:
Doh! Interestingly, on my machine I never ever hit this condition. The metadata server always seems to win the race. We potentially still have a problem if the process doesn't terminate but also never prints the line, even with up-front registration. Will fix!